### PR TITLE
feat: add capability to handle http requests with repeated slashes

### PR DESCRIPTION
**Pull request by dxfg**
WIP PR to add capability to handle http requests w/ repeated slashes.

### CODE REVIEW

**Review by rjzak — APPROVED** [review]
Looks good to me

File: crates/server/src/handle.rs [hunk 1]
```diff
@@ -109,12 +121,16 @@ pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
...
         (Some("_tag"), Some(tag), prop @ (None | Some("tree"))) => {
-            let tag = tag.parse::<TagName>().map_err(|e| {
-                (
-                    StatusCode::BAD_REQUEST,
-                    format!("Failed to parse tag name: {e}"),
-                )
-            })?;
+            let tag = tag
+                .trim_start_matches('/')
+                .trim_end_matches('/')
```
Comment (rvolosatovs) [inline crates/server/src/handle.rs:126]: This is redundant due to the `split` and the `filter`. You'll never get slashes here.
BTW, just FYI, `trim_matches` trims from both sides

File: crates/server/src/handle.rs [hunk 2]
```diff
@@ -154,3 +173,58 @@ pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
...
     }
 }
+
+#[async_std::test]
+async fn multiple_slashes_missing() {
+    let request = Request::builder()
+        .uri("https://www.rust-lang.org/")
+        .header("User-Agent", "my-awesome-agent/1.0")
+        .body(hyper::Body::empty());
+    println!("{:?}", request);
```
Comment (rvolosatovs) [inline crates/server/src/handle.rs:183]: Redundant print statement

File: crates/server/src/handle.rs [hunk 3]
```diff
@@ -154,3 +173,58 @@ pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
...
+
+    let res = handle(request.unwrap()).await;
+
+    // Temporary print to ensure test is working as intended.
+    // println!("{}", res.into_response().status());
```
Comment (rvolosatovs) [inline crates/server/src/handle.rs:188]: Avoid commented-out code

File: crates/server/src/handle.rs [hunk 4]
```diff
@@ -154,3 +173,58 @@ pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
...
+    assert_eq!(res.into_response().status(), 404);
+    return ();
+}
```
Comment (rvolosatovs) [inline crates/server/src/handle.rs:230]: If we want to have unit tests in this file, they have to reside in `mod tests`, see https://doc.rust-lang.org/book/ch11-01-writing-tests.html

It could also be a better idea to write a test for this in https://github.com/profianinc/drawbridge/blob/f03fb5815f28f5196ff4c93c4137791dc6aab67b/tests/mod.rs instead.
We could just add a few direct URL calls there, something like https://github.com/profianinc/drawbridge/blob/1871d2a7f2519c835818e33a9ca6074e67109006/crates/app/tests/helpers/tree.rs, except with additional slashes.

Basically we need to just test at least one URL, something like: https://store.profian.com//api////v0.2.0////examples//////fibonacci-rust////_tag///0.2.0//tree//PATH/////IN/TREE

if that works, we can assume that everything works.
This could be added to the end of the integration test, where the tree already exists.

File: crates/server/src/handle.rs [hunk 5]
```diff
@@ -154,3 +173,58 @@ pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
...
+
+    let res = handle(request.unwrap()).await;
+
+    // Temporary print to ensure test is working as intended.
+    // println!("{}", res.into_response().status());
+    assert_eq!(res.into_response().status(), 404);
+
+    let request = Request::builder()
+        .uri("https://www.rust-lang.org///")
+        .header("User-Agent", "my-awesome-agent///1.0")
```
Comment (rvolosatovs) [inline crates/server/src/handle.rs:193]: This is unnecessary

**Review by rvolosatovs — CHANGES_REQUESTED** [review]
implementation looks overall, but I don't see how the tests are relevant for the task at hand.

You should be using the actual URLs that this function handles, e.g. https://store.profian.com//api////v0.2.0////examples//////fibonacci-rust////_tag///0.2.0//tree///PATH//IN/TREE

**Comment by bstrie** [comment]
Status: waiting for review from Roman to be addressed.
